### PR TITLE
Enable caching using memcached

### PR DESCRIPTION
**Pull request by kajinamit**
This introduces the capability to use memcached as cache backend, to reproduce the setting in TripleO.

Enabling caching in heat provides large performance benefit in token validation and some resource queries.

### CODE REVIEW

**Comment by kajinamit** [comment]
We have to wait until https://github.com/openstack-k8s-operators/lib-common/pull/298 is merged ...

File: controllers/heat_controller.go [hunk 1]
```diff
@@ -236,6 +241,35 @@ func (r *HeatReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		return nil
 	}
 
+	memcachedFn := func(o client.Object) []reconcile.Request {
+		result := []reconcile.Request{}
+
+		// get all Horizon CRs
```
[the remark this answers is not in the document]
Comment (kajinamit) [reply controllers/heat_controller.go]: Good catch. Fixed.

**Review by bshephar — APPROVED** [review]

**Review by bshephar — COMMENTED** [review]
Would you mind updating the README with a note about this being required? We probably need some kind of automation to update the samples I used in the README. They're probably all outdated now anyway.

https://github.com/openstack-k8s-operators/heat-operator/blob/main/README.md

**Comment by kajinamit** [comment]
> Would you mind updating the README with a note about this being required? We probably need some kind of automation to update the samples I used in the README. They're probably all outdated now anyway.
> 
> https://github.com/openstack-k8s-operators/heat-operator/blob/main/README.md

Good catch. I've updated README.
I'm wondering if we can create a sample file and import it there.

**Review by bshephar — APPROVED** [review]

**Pull request merged**